src/main/java/io/github/hildi/can/service/JavaSerializationService.java

```diff
@@ -0,0 +1,239 @@
+package io.github.hildi.can.service;
+
+import io.github.hildi.can.exceptions.InvalidFileException;
+import io.github.hildi.can.exceptions.NotReadableFileException;
+import io.github.hildi.can.exceptions.NotWritableFileException;
+import io.github.hildi.can.exceptions.SerializationException;
+import io.github.hildi.can.model.FullName;
+import io.github.hildi.can.model.User;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Created by Serhii Hildi on 26.03.19.
+ */
+public class JavaSerializationService implements SerializationService{
+
+    private User user;
+    private FullName fullName;
+    private long userId;
+    private String userNickName;
+    private String firstName;
+    private String lastName;
+    private String userEmail;
+    private Collection<String> userPermissions;
+    private Map<String, String> userAttributes;
+    private Date createdAt;
+
+    @Override
+    public void serialize(User user, File file) {
+        assertParamNotNull(user, file);
+        assertFileExist(file);
+        assertFileWritable(file);
+        try (FileWriter writer = new FileWriter(file)) {
+            writer.write(setUserParametersToFile(user));
+        } catch (IOException e) {
+            throw new SerializationException("Failed to serialize data. " + file.getName(), e);
+        }
+    }
+
+    @Override
+    public User deserialize(File file) {
+        assertFileNotNull(file);
+        assertFileReadable(file);
+        assertFileExtensionIsCorrect(file);
+        try (Scanner scanner = new Scanner(file)){
+            getAndAssignParametersFromFile(scanner);
+            setUserParameters(userId, userNickName, userEmail, userPermissions, userAttributes, createdAt);
+        } catch (FileNotFoundException e) {
+            throw new SerializationException("Failed to deserialize data, reason: file " + file.getName() +
+                " is not found. ", e);
+        }
+        return user;
+    }
+
+    private String setUserParametersToFile(User user) {
+        return "id = " + user.getId() + "\n" +
+            "nickName = " + user.getNickName() + "\n" +
+//            "firstName = " + fullName.getFirstName() + "\n" +
+//            "lastName = " + fullName.getLastName() + "\n" +
+            "email = " + user.getEmail() + "\n" +
+            "permissions = " + user.getPermissions() + "\n" +
+            "attributes = " + user.getAttributes() + "\n" +
+            "createdAt = " + user.getCreatedAt() + "\n";
+    }
+
+    private void getAndAssignParametersFromFile(Scanner scanner) {
+        while (scanner.hasNextLine()) {
+            getUserPropertiesFromFile(scanner);
```

Comment:
do you need the whole scanner instance inside your `getUserPropertiesFromFile` method?
it's better to get next line right here and pass a simple string inside